aqt/metadata.py

```diff
@@ -641,23 +615,48 @@ def describe_filters(self) -> str:
             return str(self.archive_id)
         return "{} with minor version {}".format(self.archive_id, self.filter_minor)
 
-    def print_suggested_follow_up(self, printer: Callable[[str], None]) -> None:
-        """Makes an informed guess at what the user got wrong, in the event of an error."""
-        base_cmd = "aqt {0.category} {0.host} {0.target}".format(self.archive_id)
-        if self.archive_id.extension:
-            msg = "Please use '{} --extensions <QT_VERSION>' to list valid extensions.".format(
-                base_cmd
-            )
-            printer(msg)
-
-        if self.archive_id.is_tools() and self.request_type == "tool variant names":
-            msg = "Please use '{}' to check what tools are available.".format(base_cmd)
-            printer(msg)
-        elif self.filter_minor is not None:
-            msg = "Please use '{}' to check that versions of {} exist with the minor version '{}'".format(
-                base_cmd, self.archive_id.category, self.filter_minor
-            )
-            printer(msg)
-        elif self.request_type in ("architectures", "modules", "extensions"):
-            msg = "Please use '{}' to show versions of Qt available".format(base_cmd)
-            printer(msg)
+
+def suggested_follow_up(meta: ListCommand) -> str:
+    """Makes an informed guess at what the user got wrong, in the event of an error."""
+    base_cmd = "aqt {0.category} {0.host} {0.target}".format(meta.archive_id)
+    if meta.archive_id.extension:
+        msg = "Please use '{} --extensions <QT_VERSION>' to list valid extensions.".format(
+            base_cmd
+        )
+        return msg
```

Comment:
You've made a control flow change here, and I'm not sure you meant to.

The previous implementation was capable of printing multiple suggestions for the same command: one suggestion to print available extensions, and one suggestion that depends on the request type.

For example: a user types `aqt list qt6 windows android --extension x86 --modules 6.2.2`. This command will fail, because https://download.qt.io/online/qtsdkrepository/windows_x86/android/qt6_622_x86/Updates.xml does not exist.  Unless we read the html page at https://download.qt.io/online/qtsdkrepository/windows_x86/android/, we cannot know if the user got the extension wrong or the version wrong. My solution was to print both suggestions: `aqt list --extensions`, and `aqt list` to print available versions. This change would only print the `aqt list --extensions` message, and that happens to be the wrong suggestion in this case.

It now occurs to me that the right solution to this problem might be to go read the html page and figure out which part(s) the user got wrong, and then print to the console "The version 6.2.2 doesn't exist; these are the versions available: 6.0.0 ...", instead of "Try this command to list versions available".

Reply:
Now I wish I had added tests for this function!

Reply:
I don't intend to change a behavior. It would be some mistakes here.

Reply:
I don't think you have to throw out your changes entirely. I like the idea of `suggested_follow_up` returning a string that the caller could print; it just needs to be able to append multiple suggestions to the return value.

Reply:
I've reverted the function.